### PR TITLE
[SDK-1121][Docs] Add support for Braze Info.plist dictionary

### PR DESCRIPTION
**Pull request by mattlsteinberg**
# Pull Request/Issue Resolution

https://jira.braze.com/browse/SDK-1121

**Description of Change:**
An upcoming release will allow iOS SDK integrators to use the `Braze` dictionary in their `Info.plist` for configuration. `Appboy` was previously the name of this dictionary and remains silently supported.
Updating docs to reflect the recommended `Braze` naming.

### Is this change associated with a Braze feature/product release?
- [x] Yes (iOS SDK 4.0.2, April 5, 2021)
- [ ] No

---
---

<details>
<summary>PR Checklist</summary>

- [x] Check that all links work!
- [x] Ensure you have completed [our CLA](https://www.braze.com/docs/cla/).
- [ ] Tag @Timothy-Kim and @KellieHawks as a reviewer when your work is _done and ready to be reviewed for merge_.
- [x] Tag others as Reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `config/nginx.conf.erb`.

</details>

<details>
<summary>Writing & Style Quick Tips</summary>

- Use the formatting shown on the [Styling Test Page](https://github.com/Appboy/braze-docs/blob/develop/_docs/_home/styling_test_page.md) to create unique templates and vehicles for your information.
- Use Action verbs!
- Capitalize proper nouns, unique Braze tools and features, custom Braze channels, and all Header titles.
- Redirect when necessary!
- Always speak to the Braze customer - refer to them as “you”, whether they are an Engineer/Developer, a Marketer, or an Admin.
- On Voice: You should aim to be a Friendly Professor. Imagine, if you will, the smartest, but also kindest professor you had. You know, the one who showed you your potential and tested you, but also went slowly in class so you could take great notes and learn some things? Don't be too peppy, but don't be formal. You’re not selling anything - you’re instructing and informing and enabling.

_Consult the [Docs Text Formatting Guide](https://github.com/Appboy/braze-docs/wiki/Special-Formatting) and the [Docs Writing Style Guide & Best Practices](https://github.com/Appboy/braze-docs/wiki/Writing-Style-Guide-&-Best-Practices) for more details._

</details>

<details>
<summary>ATTENTION: For PR Reviewers</summary>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Preview all changes in the linked Heroku environment (click `View deployment` button below, then "Docs". A `502` error just means you should refresh until you see the Docs Home page.
</details>

---
---

### CODE REVIEW

**Review by lowip — DISMISSED** [review]
lgtm

File: _docs/_developer_guide/eu01_us3_sdk_implementation_differences/for_ios.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Note that as of December 2019, custom endpoints are no longer given out, if you
 {% endalert %}
 
 If given a pre-exisiting custom endpoint by Braze...
-- Starting with Braze iOS SDK v3.0.2, you can set a custom secure endpoint using the `Info.plist` file. Add the `Appboy` dictionary to your Info.plist file. Inside the `Appboy` dictionary, add the `Endpoint` string subentry and set the value to your custom endpoint url’s authority (for example, `sdk.iad-01.braze.com`, not `https://sdk.iad-01.braze.com`).
+- Starting with Braze iOS SDK v3.0.2, you can set a custom secure endpoint using the `Info.plist` file. Add the `Braze` dictionary to your Info.plist file. Inside the `Braze` dictionary, add the `Endpoint` string subentry and set the value to your custom endpoint url’s authority (for example, `sdk.iad-01.braze.com`, not `https://sdk.iad-01.braze.com`).
```
Comment (Bucimis) [inline _docs/_developer_guide/eu01_us3_sdk_implementation_differences/for_ios.md:17]: We should add a note to each instance stating something like "Note that prior to Braze iOS SDK x.y.z, the dictionary key `Appboy` must be used in place of `Braze`."

**Review by Bucimis — DISMISSED** [review]
Thanks for writing this up! We should remember not to merge until release.

File: _docs/_developer_guide/eu01_us3_sdk_implementation_differences/for_ios.md (hunk 1 above)
Comment (mattlsteinberg) [reply _docs/_developer_guide/eu01_us3_sdk_implementation_differences/for_ios.md:17]: Noted, will add once we know what X, Y, and Z are :)

**Review by chshapiro — APPROVED** [review]

**Review by hokstuff — APPROVED** [review]

**Comment by KellieHawks** [comment]
Hello @mattlsteinberg, just following up. Is this PR ready to be merged?

**Comment by Bucimis** [comment]
@KellieHawks not yet, thanks for checking

File: _docs/_developer_guide/eu01_us3_sdk_implementation_differences/for_ios.md (hunk 1 above)
Comment (mattlsteinberg) [reply _docs/_developer_guide/eu01_us3_sdk_implementation_differences/for_ios.md]: Looks like it's going to be 4.0.2, so I've added this note to each instance. Will wait for release before requesting this to be merged.

**Comment by mattlsteinberg** [comment]
> Hello @mattlsteinberg, just following up. Is this PR ready to be merged?

@KellieHawks this is ready to merge! Part of the recent iOS SDK 4.0.2 release.

**Review by KellieHawks — APPROVED** [review]

**Pull request merged**